editor/src/core/shared/either.ts

```diff
@@ -119,6 +119,14 @@ export function eitherToMaybe<L, R>(either: Either<L, R>): R | null {
   }
 }
 
+export function maybeEitherToMaybe<L, R>(either: Either<L, R> | null | undefined): R | null {
```

Comment:
I feel a little uncomfortable with this, it's a little _too_ custom to be something that should be in here.

Reply:
nice catch, I actually wanted to ping you about this! 

The problem is that without this it is very hard to write concise code with `?.` operators. Especially if the eitherToMaybe is in the middle of an expression, it forces you to break it out into a ternary.

But I would be happy if you can come up with a nicer solution.

This I think is a good example of code that `maybeEitherToMaybe` makes possible:
`const componentPackageName = maybeEitherToMaybe(componentMetadata?.importInfo)?.path`

The closest alternative I guess would be
`const componentPackageName = optionalMap(eitherToMaybe, componentMetadata?.importInfo)?.path`

but at least for me this is just over the threshold and crosses into hard-to-parse-at-a-glance territory

Reply:
For me, it feels like a lens'y thing that you're asking for.

```
maybeEitherToMaybe :: Maybe (Either a b) -> Maybe b
maybeEitherToMaybe = firstOf (_Just . _Right)
```

In that the `_Just . _Right` is the lens.

Reply:
There's an argument that this `maybeEitherToMaybe(componentMetadata?.importInfo)?.path` is a little too dense to read, because it's optionally getting a field, transforming that and if that result is available getting a field within that value.

Reply:
This to me definitely feels like it's falling into the realm of trying to achieve too much in a single line of code. All of these are because `ElementInstanceMetadata.element` is of type `Either<string, JSXElementChild>`, and I'm now wondering if that even makes sense anymore? If that does still make sense, then probably a clearer way to handle this would be to add functions to `element-metadata-utils` for e.g. `getComponentPackageNameFromMetadata`